tests/kvm_host_ceph-nautilus.nix

```diff
@@ -99,39 +102,29 @@ let
           (pkgs.writeShellScriptBin "run-tests" # BEWARE: DO NOT RENAME!
               ''
             set -o pipefail
-            set -x
 
             export PYTHONPATH="${PYTHONPATH}"
             export PATH="${PATH}:${pkgs.openssh}/bin:${pkgs.gnused}/bin"
+            export PYTHONUNBUFFERED=1
+
             cd ${testPackage.src}
 
-            rm /tmp/fc.qemu-report.xml
+            rm -f /tmp/fc.qemu-report.xml
 
-            pytest -vv --cov-config=/etc/coveragerc --cov-append -c ${testPackage.src}/pytest.ini "$@" 2>&1 | tee /dev/kmsg
-            PYTESTRET=$?
-
-            ${if testOpts != "" then ''
-            # If we run with custom test options we might be filtering
-            # for tests and due to the live/not live split either phase
-            # might not have any tests. However, we do not want to
-            # accidentally accept the `no tests found` failure in Hydra.
-            echo "Pytest result code: $PYTESTRET"
-            if [ $PYTESTRET -eq 0 ] || [ $PYTESTRET -eq 5 ]; then
-              # 5 means no tests found, which might happen if we have options
-              true;
-            fi
-            '' else ""}
+            pytest -vv --cov-config=/etc/coveragerc --cov-append -c ${testPackage.src}/pytest.ini "$@" | tee /dev/console
 
+            # We're not looking at the exit code of the above pipe as the
+            # tee to /dev/kmsg sometimes causes a non-zero exit code ...
```

Comment:
```suggestion
            # tee to /dev/console sometimes causes a non-zero exit code ...
```

Reply:
Aeh nope ... /dev/kmsg causes issues. /dev/console is fine. However, the comment is now superfluous. I still think looking at the report file is a good idea.